rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceWindow.java

```diff
@@ -0,0 +1,473 @@
+
+        nodeSourceWindowLayout.addMember(nodeSourceWindowLabel);
+        nodeSourceWindowLayout.addMember(nodeSourceWindowForm);
+        nodeSourceWindowLayout.addMember(scrollLayout);
+        nodeSourceWindowLayout.addMember(buttonsLayout);
+
+        int winWidth = com.google.gwt.user.client.Window.getClientWidth() * 80 / 100;
+        int winHeight = com.google.gwt.user.client.Window.getClientHeight() * 80 / 100;
+        winWidth = Math.min(1000, winWidth);
+        winHeight = Math.min(1000, winHeight);
+
+        this.window = new Window();
+        this.window.setTitle(this.windowTitle);
+        this.window.setShowMinimizeButton(false);
+        this.window.setIsModal(true);
+        this.window.setShowModalMask(true);
+        this.window.addItem(nodeSourceWindowLayout);
+        this.window.setWidth(winWidth);
+        this.window.setHeight(winHeight);
+        this.window.setCanDragResize(true);
+        this.window.setCanDragReposition(true);
+        this.window.centerInPage();
+    }
+
+    private void prepareCreateAndDeployFormAndSubmit(VLayout nodeSourceWindowLayout,
+            Label nodeSourcePluginsWaitingLabel, DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel,
+            TextItem nodeSourceNameItem, CheckboxItem nodesRecoverableItem, List<IButton> buttonList,
+            boolean nodeSourceEdited) {
+
+        nodeSourcePluginsForm.setValue("deploy", Boolean.TRUE.toString());
+
+        prepareCreateOnlyFormAndSubmit(nodeSourceWindowLayout,
+                                       nodeSourcePluginsWaitingLabel,
+                                       nodeSourcePluginsForm,
+                                       nodeSourceWindowLabel,
+                                       nodeSourceNameItem,
+                                       nodesRecoverableItem,
+                                       buttonList,
+                                       nodeSourceEdited);
+    }
+
+    private void prepareCreateOnlyFormAndSubmit(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
+            DynamicForm nodeSourcePluginsForm, Label nodeSourceWindowLabel, TextItem nodeSourceNameItem,
+            CheckboxItem nodesRecoverableItem, List<IButton> buttonList, boolean nodeSourceEdited) {
+
+        nodeSourcePluginsForm.setValue("infra", this.infrastructureSelectItem.getValueAsString());
+        nodeSourcePluginsForm.setValue("nsName", nodeSourceNameItem.getValueAsString());
+        nodeSourcePluginsForm.setValue("nodesRecoverable", nodesRecoverableItem.getValueAsBoolean().toString());
+        nodeSourcePluginsForm.setValue("policy", this.policySelectItem.getValueAsString());
+        nodeSourcePluginsForm.setValue("sessionId", LoginModel.getInstance().getSessionId());
+        nodeSourcePluginsForm.setValue("nodeSourceEdited", Boolean.toString(nodeSourceEdited));
+        nodeSourcePluginsForm.setCanSubmit(true);
+
+        /*
+         * this smartGWT form looks nice but cannot do callbacks ;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)